cmd/install/assets/hypershift_operator.go

```diff
@@ -316,6 +320,11 @@ func (o ExternalDNSDeployment) Build() *appsv1.Deployment {
 		)
 	}
 
+	// Add proxy settings if cluster has a proxy
+	if o.Proxy != nil {
```

Comment:
Should you check for Status too? Looks like it's optional - https://github.com/openshift/hypershift/blob/e93a5e12c668b997519220e14db5798f008bcfa3/vendor/github.com/openshift/api/config/v1/types_proxy.go#L34.

Reply:
It is optional, but it can't be nil (not a pointer).  `SetEnvVarsTo` does the Right Thing if the fields are empty strings.